File: src/Enterprise/C#/patterns/DistributedProxy.cs
namespace Genkidama.PatternExamples;
public static class DistributedProxyExample { public static bool Run(){int Remote(string sku)=>sku=="sku-1"?7:0;int Proxy(string sku)=>Remote(sku);return Proxy("sku-1")==7;} }

File: src/Enterprise/VB.NET/patterns/DistributedProxy.vb
Imports System
Friend Module DistributedProxyExample
    Friend Function Run() As Boolean
        Dim remote As Func(Of String,Integer)=Function(sku) If(sku="sku-1",7,0)
        Dim proxy As Func(Of String,Integer)=Function(sku) remote(sku)
        Return proxy("sku-1")=7
    End Function
End Module
